core/src/CMakeLists.txt

```diff
@@ -160,7 +160,9 @@ KOKKOS_LINK_TPL(kokkoscore PUBLIC LIBRT)
 IF (NOT WIN32)
   KOKKOS_LINK_TPL(kokkoscore PUBLIC THREADS)
 ENDIF()
-KOKKOS_LINK_TPL(kokkoscore PUBLIC ROCM)
+IF (NOT KOKKOS_ENABLE_COMPILE_AS_CMAKE_LANGUAGE)
+  KOKKOS_LINK_TPL(kokkoscore PUBLIC ROCM)
```

Comment:
@crtrott not sure it will play nice with #5192 ?

Reply:
I guess for #5192 it would just need similar `IF` statements around the `ROCTHRUST` TPL being introduced

Reply:
Or actually maybe not, if `ROCTHRUST` is still something you need to find even if you're using the HIP compiler

Reply:
I'm fine merging both and then I'll fix it again with my code when they're both in `develop`

Reply:
I would actually think its orthogonal.